File: src/app/models/vacancy.ts
import { Project } from "./project";
import { Candidate } from "./candidate";
import { SkillVacancy } from "./skill-vacancy";


export class Vacancy {
    vacancyId: number;
    vacancyTitle: string;
    dueDate: Date;
    skillVacancy: SkillVacancy[];
    candidates: Candidate[];
    projectKey: number;
    project: Project;
}

File: DbModel/Vacancy.cs
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System;
using Microsoft.EntityFrameworkCore.Metadata;

namespace Ng2Core.CorpPortal.Models
{
    public class Vacancy
    {
        [Key]
        public int VacancyId { get; set; }
		[MaxLength(200)]
        public string VacancyTitle { get; set; }
		[Column(TypeName="date")]
        public DateTime DueDate { get; set; }

        [InverseProperty("Vacancy")]
        public List<SkillVacancy> SkillVacancy { get; set; }

        [InverseProperty("Vacancy")]
        public List<Candidate> Candidates { get; set; }

        public int ProjectKey { get; set; }
        [ForeignKey("ProjectKey")]
        public Project Project { get; set; }
    }
}